go/vt/tabletserver/tabletserver.go

```diff
@@ -176,6 +181,63 @@ func NewTabletServer(config Config) *TabletServer {
 	return tsv
 }
 
+// createTransactionThrottlerOrPanic creates the transaction throttler from the configuration
+// given in the relevant fields in tsv.config. It does nothing if the transaction throttler is not
+// enabled (tsv.config.EnableTxThrottler is false).
+// It panics if any error occurs or if the tabletserver already has a transaction throttler.
+func (tsv *TabletServer) createTransactionThrottlerOrPanic() {
+	if !tsv.config.EnableTxThrottler {
+		return
+	}
+	if tsv.transactionThrottler != nil {
+		panic("Transaction throttler already exists!")
+	}
+	var throttlerConfig throttlerdatapb.Configuration
+	err := proto.UnmarshalText(
+		tsv.config.TxThrottlerConfig,
+		&throttlerConfig)
+	if err != nil {
+		panic(fmt.Errorf("Can't parse TxThrottlerConfig: %v", err))
+	}
+
+	healthCheckCells := parseTransactionThrottlerCellsOrPanic(tsv.config.TxThrottlerHealthCheckCells)
+	transactionThrottler, err := NewTransactionThrottler(
+		&TransactionThrottlerConfig{
+			Keyspace:        tsv.target.Keyspace,
+			Shard:           tsv.target.Shard,
+			ThrottlerConfig: &throttlerConfig,
+
+			HealthCheckCells: healthCheckCells,
+		})
+	if err != nil {
+		panic(fmt.Errorf("Can't create transaction throttler: %v", err))
+	}
+	tsv.transactionThrottler = transactionThrottler
+}
+
+func parseTransactionThrottlerCellsOrPanic(cellsCSV string) []string {
+	csvReader := csv.NewReader(strings.NewReader(cellsCSV))
+	cells, err := csvReader.Read()
+	if err != nil {
+		panic(fmt.Errorf("Can't parse transaction throttler health-check cells specification. "+
+			"It should be a CSV string. Got: %v", cellsCSV))
+	}
+	return cells
+}
+
+// destroyTransactionThrottler destroys and the throttler and deallocates
+// all its resources. It's ok to call this function more than once.
+func (tsv *TabletServer) destroyTransactionThrottler() {
+	if !tsv.config.EnableTxThrottler {
+		return
+	}
+	if tsv.transactionThrottler == nil {
+		return
+	}
+	tsv.transactionThrottler.Close()
+	tsv.transactionThrottler = nil
+}
+
```

Comment:
The model for TabletServer `services` is that you always create the object as `NewXXX` in `NewTabletServer`, and then `Open` and `Close` as needed. This way, the service can export `/debug/vars` even if it's closed.
Here's an example of how I refactored update stream: https://github.com/youtube/vitess/pull/2412.